cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt

```diff
@@ -394,95 +395,102 @@ class StatementHandler(frontend: PythonLanguageFrontend) :
+        positionalArguments: List<Python.ASTarg>,
+        args: Python.ASTarguments,
+        result: FunctionDeclaration,
+        recordDeclaration: RecordDeclaration
+    ) {
+        // first argument is the receiver
+        val recvPythonNode = positionalArguments.firstOrNull()
+        if (recvPythonNode == null) {
             val problem =
                 newProblemDeclaration(
-                    "`kw_defaults` are not yet supported",
+                    "Expected a receiver",
                     problemType = ProblemNode.ProblemType.TRANSLATION,
                     rawNode = args
                 )
             frontend.scopeManager.addDeclaration(problem)
+        } else {
+            val tpe = recordDeclaration.toType()
+            val defaultValue =
+                args.defaults.getOrNull(0)?.let { frontend.expressionHandler.handle(it) }
+            val recvNode =
+                newVariableDeclaration(
+                    name = recvPythonNode.arg,
+                    type = tpe,
+                    implicitInitializerAllowed = false,
+                    rawNode = recvPythonNode
+                )
+            if (defaultValue != null) {
+                val problem =
+                    newProblemDeclaration(
+                        "Receiver with default value",
+                        problemType = ProblemNode.ProblemType.TRANSLATION,
+                        rawNode = args
+                    )
+                frontend.scopeManager.addDeclaration(problem)
+            }
+            frontend.scopeManager.addDeclaration(recvNode)
+            when (result) {
+                is ConstructorDeclaration -> result.receiver = recvNode
+                is MethodDeclaration -> result.receiver = recvNode
+                else -> TODO()
+            }
         }
+    }
 
-        args.kwarg?.let {
-            val problem =
-                newProblemDeclaration(
-                    "`kwarg` is not yet supported",
-                    problemType = ProblemNode.ProblemType.TRANSLATION,
-                    rawNode = it
-                )
-            frontend.scopeManager.addDeclaration(problem)
+    private fun handlePositionalArguments(
+        positionalArguments: List<Python.ASTarg>,
+        args: Python.ASTarguments,
+        startIdx: Int
+    ) {
+        val nonDefaultArgsCount = positionalArguments.size - args.defaults.size
+
+        for ((idx, arg) in positionalArguments.drop(startIdx).withIndex()) {
+            val defaultIndex = idx - nonDefaultArgsCount
+            val defaultValue =
+                if (defaultIndex >= 0) {
+                    args.defaults.getOrNull(defaultIndex)?.let {
+                        frontend.expressionHandler.handle(it)
+                    }
+                } else {
+                    null
+                }
+            handleArgument(
+                arg,
+                isPosOnly = arg in args.posonlyargs,
+                isKwoOnly = defaultValue != null,
```

Comment:
is this `isKwoOnly = defaultValue != null,` obvious? otherwise: please some doc

Reply:
Since the property “isKwoOnly” is a boolean expression, the result i.e. if the condition is true, returns true otherwise false

Reply:
I'm still confused. The method is called `handlePositionalArguments` and this code reads like the _positional_ argument can be _keyword only_? I need to read up on this.

Reply:
Thinking more about it: Don't the keyword and positional arguments affect the function calls more than the declarations? Does this "only" require fixing the SymbolResolver or also an update of the frontend?

I think that I can always use the keyword syntax even for positional arguments. Maybe the "only" makes the difference here which is somewhat confusing and it should be "isKeyword" and "isPositional" and then, there might be a combination where everything can be used in the function call

Reply:
As we are handling not only the posonlyargs but also the "normal" args in this method and these can either be kwonlyargs or not, I removed the property isKwOnly here.